deepcell/model_zoo/panopticnet.py

```diff
@@ -326,6 +326,15 @@ def PanopticNet(backbone,
                              'from {}.'.format(temporal_mode,
                                                str(acceptable_modes)))
 
+    
+    # TODO only works for 2D: do we check for 3D as well? What are the requirements for 3D data?
+    img_shape = input_shape[1:] if channel_axis == 1 else input_shape[:-1]
+    if img_shape[0] != img_shape[1]:
+        raise ValueError('Input data must be square, got dimensions {}'.format(img_shape))
+
+    if img_shape[0] not in [2 ** x for x in range(5, 12)]:
+        raise ValueError('Input data dimensions must be a power of 2, got {}'.format(img_shape[0]))

```

Comment:
Why not just use `log` to check? That would remove the need for the size limits and would be O(1) instead of O(N).

```python
if math.log(img_shape[0], 2).is_integer():
  ...
```